apps/search/js/providers/grid_provider.js

```diff
@@ -21,8 +23,15 @@ DataGridProvider.prototype = {
 
   render: function(results) {
     results.forEach(function(config, index) {
+      if (this.grid.getLength() >= this.MAX_RESULTS) {
```

Comment:
I agree it would be better not to touch the component, but GridProvider could only count the results for one provider, the grid gets results from multiple providers which return asynchronously. I tried a few ways of doing this, and this was the most reliable.